src/pyclaw/solution.py

```diff
@@ -332,19 +334,38 @@ def read(self, frame, path='./_output', file_format='ascii',
         :Output:
          - (bool) - True if read was successful, False otherwise
         """
-        
-        read_func = self.get_read_func(file_format)
 
         path = os.path.expandvars(os.path.expanduser(path))
+
+        # Attempt to determine the type of file - Note that this will NOt detect
+        # whether ForestClaw should be used.
+        if file_format is None:
+            for root, dirs, files in os.walk(path):
+                for file_name in files:
+                    if file_name.startswith(file_prefix + ".b"):
+                        file_format = "binary"
+                        break
+                    elif file_name.endswith(".hdf"):
+                        file_format = "hdf"
+                        break
+                    elif file_name.endswith(".ptc"):
+                        file_format = "petsc"
+                if file_format is None:
```

Comment:
I'm wondering if it would be preferable to check for the presence of ascii files, and if they're not present either then throw an error saying that no output files were found.